src/DataStream/Contouring.cc

```diff
@@ -234,9 +235,8 @@ void TraceContours(float* image, int64_t width, int64_t height, double scale, do
     }
 
     if (spdlog::get(PERF_TAG)) {
-        auto t_end_contours = std::chrono::high_resolution_clock::now();
-        auto dt_contours = std::chrono::duration_cast<std::chrono::microseconds>(t_end_contours - t_start_contours).count();
-        auto rate_contours = width * height / (double)dt_contours;
+        auto dt_contours = t.Elapsed(Timer::ms);
+        auto rate_contours = width * height / t.Elapsed(Timer::us);
```

Comment:
Surely every time you call `Elapsed` you recalculate the value using a different end time? This is why I was suggesting returning a struct from `Elapsed`:

```
auto dt = t.Elapsed();
auto dt_contours = dt.ms;
auto dt_rate_contours = width * height / dt.us;
```

The `TimeDelta` struct constructor can set the `ms` and `us` properties from a `us` parameter.

(Since the `ms` value is derived from the original `us` value, and we use it in every instance that we use a timer here, I don't think it's necessary to do anything more complicated to avoid calculating it every time we construct a `TimeDelta`.)